CMakeLists.txt

```diff
@@ -82,6 +82,7 @@ message(STATUS "Optimization level: ${OPTIMIZE}")
 
 if(MSVC)
   # Microsoft Visual Studio Compiler
+  add_compile_options(/UTF8)
```

Comment:
I think it suffices to add it to the mixxx-lib target. Let's not modify the command line for everything in the lib directory.

Reply:
We do that for plenty of other compile options. I don't think there is a harm in doing so. Also, I think this may be useful for our tests.

Reply:
> We do that for plenty of other compile options. I don't think there is a harm in doing so.

There might be conflicting command lines. Using non-target-based cmake commands is considered bad practice.
We should definitely refactor that as some point.

> Also, I think this may be useful for our tests.

What do you mean? mixxx-lib is linked again mixxx-test, so they would use that option too?

Reply:
> What do you mean? mixxx-lib is linked again mixxx-test, so they would use that option too?

I don't think compiler options for libraries automatically get applied to targets the libraries are linked to.

Reply:
> Using non-target-based cmake commands is considered bad practice.
We should definitely refactor that as some point.

Currently all our compiler options are handled before mixxx-lib is defined, so that would take a bit of refactoring.

Reply:
> I don't think compiler options for libraries automatically get applied to targets the libraries are linked to.

They are, if you set them as PUBLIC. 

https://cmake.org/cmake/help/latest/command/target_compile_options.html

> Currently all our compiler options are handled before mixxx-lib is defined, so that would take a bit of refactoring.

We just could move it behind the library definition. The real issue is that options defined for mixxx-lib are not applied to targets linked to mixxx-lib. Therefore we should use `target_sources` instead of adding new targets like `mixxx-xwax`. Out of scope for this PR.

Reply:
Ping. Anything left to do?

Reply:
@Holzhaus can we merge this?

Reply:
> @Holzhaus can we merge this?

Yes